charts/istio-alerts/values.yaml

```
@@ -23,7 +23,7 @@ serviceRules:
   # -- Narrow down the alerts to a particular Destination Service if there
   # are multiple services that require different thresholds within the same
   # namespace.
-  destinationServiceName: .*
+  destinationServiceName: ""
```

If I understand this correctly, combined with the if/else change in _helpers.tpl, this will add the newly created istio-alerts alias into the destination_service_name selector for the query...

Unless, I'm misunderstanding the change, this is _not_ what we'd want, right? (because `destination` for any of these workloads can be anything).... so my understanding is keep the destination selector changes as they were (?)

(again, want to underscore, I may be missing some context here so please correct me).

---

So, there is no way for the istio-alerts to knwo the exact name of the service, but by default it uses the chart-name which is the alias as the service name. At least based on the few services that I looked at.
Using that as a fallback is better than using ".*" which means the same alert will fire multiple times due to the multiple prometheus rules.
An alternative - we could require to fail any time "destinationService" is not set, but that means failing all our deployments until people change it - not sure there will be a go for such breaking change.